descartes_moveit/src/ikfast_moveit_state_adapter.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2016, Jonathan Meyer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "descartes_moveit/ikfast_moveit_state_adapter.h"
+
+#include <cassert>
+#include <eigen_conversions/eigen_msg.h>
+#include <ros/node_handle.h>
+
+const static std::string default_base_frame = "base_link";
+const static std::string default_tool_frame = "tool0";
+
+// Compute the 'joint distance' between two poses
```

Comment:
The L1 is correct, assuming the `distance` function is not accessible outside the class.  If it is, then distance should mean distance (L2 norm).

Efficiency is important in this class, so use of the L1 even though it isn't exactly correct is better.  Of course this assumes I'm right about the relative complexity of L1 vs L2 (given our discussion at lunch, I can't be sure).